src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -243,6 +208,50 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
         this.runByLineInProgress.set(runningByLine).ignoreErrors();
     }
 
+    private async tryToStartDebugging(mode: KernelDebugMode, editor?: NotebookEditor, cell?: NotebookCell) {
+        if (editor) {
+            if (this.notebookInProgress.has(editor.document)) {
+                return;
```

Comment:
I think returning early such as 

```
if (!editor) {
	return;
}
```

& a few of thse will improve readabiliyty of this method, right now its nested way too much.